specs/vep-1507-vdk-operations-ui/README.md

```diff
@@ -164,6 +166,190 @@ Consider at least the below topics but you do not need to cover those that are n
+
+Shows a success/failure pie chart for the total executions of the data job.
+Shows a graph of job duration over time. Graph is zoomable and the time period
+can be adjusted. Shows execution history for the job. Executions history table
+has the following columns:
+- Status
+- Type
+- Duration
+- Start (UTC)
+- End (UTC)
+- ID
+- Version
+- Logs
+
+The execution history table can be filtered by
+- Status
+- Type
+- ID
+- Version
+
+Deployment details for an individual execution can be viewed using a button
+attached to the row.
+
+**Data Job Execution Deployment Details Pop-Up**
+
+Shows the follwoing details about an individual execution
+- Version
+- Deployed by
+- Deployed at (UTC)
+- Schedule cron (UTC)
+- VDK version
+
+Shows the following resource information about an individual execution
+- CPU Limit
+- CPU Request
+- Memory Limit
+- Memory Request
+
+More info about k8s resource requests, resource limits and resource units.
+
+https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits
+
+https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes
+
+#### User Journeys
+
+**Active development**
+
+The VDK frontend empowers users to iterate quickly during the active development
+phase of new data jobs. It provides the necessary tools for developers to view
+the status and output of their jobs, as well as performance and resource data to
+help with optimization.
+
+*Example scenario 1*
+
+<!-- User creates a new job and tests it on different types of input data-->
+A user is actively developing a new data job that should be able to work with
+different data sources. They create and deploy the job using the vdk cli. The
+user then changes the data source configuration and uses the executions tab to
+view logs and check how each data source impacts job duration.
+
+*Example scenario 2*
+<!-- User has data heavy ingestion and wants to optimize their data jobs -->
+A user would like to optimize an in-development data job that uses a complex SQL
+query to ingest a large amount of data. The user has several ideas about
+optimizing the query, but is not sure which approach scales better for a
+sufficiently large amount of data. The user tests all the approaches they have
+in mind by doing code chagnes and creating consecutive data job deployments.
+They use the job duration graph on the data jobs executions details page to
+compare how their approaches scale for a sufficiently large amount of data. The
+user chooses the best approach based on their findings.
+
+*Example scenario 3*
+<!-- User creates a new job, but it fails to start -->
+A user is actively developing a new data job. They do a code change and they see
+their job has failed in the data job details page. They open the logs from the
+provided link and quickly find out what caused the error.
+
+**Monitoring**
+
+The VDK frontend empowers users to detect problems at a glance. It also provides
```

Comment:
I'd add one more scenario that I am aware of.

One of the reasons why job executions and status is visible to all (not just the team owning the job) is that it can be used by indirect users of VDK (users of data managed by a team that is using VDK). 

For example: 

Samantha, a data analyst, has received complaints about data quality issues in a recent report. She logs into the web UI to identify the data jobs which produced tables for that report with failing or delayed executions in the last 24 hours. Samantha navigates to the Home Page and checks the data jobs with failing executions widget. She clicks on the job names to go to the individual job details page and analyzes the logs and execution history . She takes note of the failing data job and the owner team responsible for it. Samantha reaches out to the job owner, shares her findings, and collaborates with them to identify the cause of the failure and resolve the issue.

Reply:
Bumping this thread up, this seems relevant. In a usual supply chain business scenario, the (multi-layered) end users may be consuming the data workloads metadata (data jobs deployments and executions) in addition to the data itself. For example, observing business requirement requests made and following on their state of progress. (Transparency.)